lib/sles4sap_publiccloud.pm

```diff
@@ -831,4 +837,105 @@ sub list_cluster_nodes {
     die 'None of the hosts are currently part of existing cluster' unless @cluster_nodes;
 }
 
+=head2 get_hana_database_info
+
+    Parses "hdbsql" command output.
+    Returns 1 if hana database is online, 0 otherwise.
+
+=back
+=cut
+
+sub get_hana_database_info {
+    my ($self, $password_db, $instance_id) = @_;
+    my $hdb_cmd = "hdbsql -u SYSTEM -p $password_db -i $instance_id 'SELECT * FROM SYS.M_DATABASES;'";
+
+    sleep 30;
+    eval {
```

Comment:
Why is this `eval` needed here? If you're using this to avoid the test from failing on an internal API failure, you can add `proceed_on_failure => 1` to the `run_cmd` call so it does not kill your test.